dart/lib/src/sentry.dart

```diff
@@ -108,9 +106,28 @@ class Sentry {
     _hub = Hub(options);
     hub.close();
 
-    // execute integrations after hub being enabled
-    for (final integration in options.integrations) {
-      await integration(HubAdapter(), options);
+    if (appRunner != null) {
+      final runZonedGuardedIntegration = options.integrations
+        .whereType<RunZonedGuardedIntegration>().first;
+      assert(
+        runZonedGuardedIntegration != null, 
+        'RunZonedGuardedIntegration should have been added when there is an appRunner.'
+      );
+
+      runZonedGuardedIntegration.runner = () async {
+        final integrations = options.integrations
+            .where((i) => i != runZonedGuardedIntegration);
+        for (final integration in integrations) {
+          await integration(hub, options);
+        }
+        await appRunner();
+      };
+
+      await runZonedGuardedIntegration(HubAdapter(), options);
+    } else {
+      for (final integration in options.integrations) {
+        await integration(HubAdapter(), options);
+      }
```

Comment:
we could create a method that takes a list of integrations and execute the integrations.
right now we are repeating both for loops